### ISSUE TITLE
Support separately-managed module services

### ISSUE THREAD

**Issue by thearchitector** [weaviate/weaviate#6048]
### Describe your feature request

I have a fairly complicated K8s cluster into which I’m deploying a multi-node Weaviate setup. I’m using custom services / deployments to underpin the `text2vec-transformers` and `reranker-transfomers` modules.

I’ve noticed that Weaviate will wait for those pods to be `Ready` before becoming `Ready` itself. It would be useful to be able to disable that, so could control my custom module services independently of Weaviate; during deployments & upgrades, I'd be able to spin up/roll Weaviate before spinning them up alongside my other deployments that actually use Weaviate. Doing would also allow me to better version my custom services, potentially as part of a separate Helm chart, without having to roll Weaviate every time I update them.

Since they'd be managed separately, I could see Weaviate either
1. coming online, then rejecting requests until the service endpoints ping healthy.
2. queuing queries until the services come online to process them, or until the request timeout is reached.

The latter seems like it could integrate really well into the existing `index_queue_*` Prometheus metrics.

### Code of Conduct

- [X] I have read and agree to the Weaviate's [Contributor Guide](https://weaviate.io/developers/contributor-guide) and [Code of Conduct](https://weaviate.io/service/code-of-conduct)

**Comment by dudanogueira** [comment]
This discussion initiated here: https://forum.weaviate.io/t/deploying-weaviate-before-modules/5732/2

**Comment by antas-marcin** [comment]
@thearchitector I have added ([PR](https://github.com/weaviate/weaviate/pull/6501)) an option of configuring the possibility of not waiting for self-deployed modules to be up and running in order to start Weaviate.

With this change all of our self-deployed modules will have their own environment variable setting available for enabling this behaviour, so in your case you would need to add 2 additional environment settings to make it to work:
- `TRANSFORMERS_WAIT_FOR_STARTUP=false`
- `RERANKER_WAIT_FOR_STARTUP=false`

also if you will send a request to Weaviate's `/v1/meta` endpoint and the self-deployed modules won't be reachable then you will will see an `error` in the response like so (this error will be replaced by a normal response once the self-deployed modules become available):
```json
{
    "hostname": "http://[::]:8080",
    "modules": {
        "reranker-transformers": {
            "error": "send GET meta request: Get \"http://reranker-transformers:8080/meta\": dial tcp 172.21.0.4:8080: connect: connection refused"
        },
        "text2vec-transformers": {
            "error": "send GET meta request: Get \"http://t2v-transformers-baai-bge-m3-onnx:8080/meta\": dial tcp 172.21.0.3:8080: connect: connection refused"
        }
    },
    "version": "1.25.27"
}
```

This feature will be available with our next patch releases: `v1.25.27`, `v1.26.12` and `v1.27.7`

**Comment by thearchitector** [comment]
@antas-marcin awesome, thank you!

i take this to mean incoming query etc requests to weaviate will throw an error as well?

**Comment by antas-marcin** [comment]
Yes, queries which use those modules that are still starting up will return error until those modules become available
